### PR TITLE
lm-sensors: update to 3.5.0

### PR DESCRIPTION
**Pull request by dengqf6**
Maintainer: @jow- 
Compile tested: mvebu
Run tested: mvebu (w/ `sensors` and `collectd`)

Description:
Update to 3.5.0
Also switch to GitHub tarball

### CODE REVIEW

**Comment by dengqf6** [comment]
I didn't bump it to latest 3.5.0 because it breaks `collectd` build. But this can actually be solved by backporting a commit from collectd master branch collectd/collectd@069796b.

**Comment by dengqf6** [comment]
Also run tested. It works fine. Anyone who wants to test can pick this branch dengqf6@65dd842

**Comment by dengqf6** [comment]
Updated to 3.5.0 with the commit patch in another branch, works fine. Anyone who wants to test can pick this branch dengqf6/packages@65dd842

**Comment by hnyman** [comment]
> I didn't bump it to latest 3.5.0 because it breaks collectd build. But this can actually be solved by backporting a commit from collectd

As the upstream patch was also in upstream 5.8 branch, I backported it here as 1f0a0057

**Comment by dengqf6** [comment]
Branch updated

**Comment by hnyman** [comment]
works for me 
mvebu WRT3200ACM temp sensors with luci-statistics & collectd:

```
 OpenWrt SNAPSHOT, r9019-0e8d5ff0fc
 -----------------------------------------------------
root@router3:~# opkg list-installed | grep sensor
collectd-mod-sensors - 5.8.1-3
libsensors - 3.5.0-1
```

**Comment by pprindeville** [comment]
Please use the standard template for the Description.

File: utils/lm-sensors/Makefile [hunk 1]
```diff
@@ -8,15 +8,14 @@
...
-PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
-PKG_MIRROR_HASH:=892e9e2eed78e27fb7a08e8ce78b1164a3d101ba79a9e04f1c1273940d48cf17
+PKG_VERSION:=3.5.0
+PKG_RELEASE:=1
+
+PKG_SOURCE_URL:=https://github.com/lm-sensors/lm-sensors/archive/V$(subst .,-,$(PKG_VERSION))/lm-sensors
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_HASH:=f671c1d63a4cd8581b3a4a775fd7864a740b15ad046fe92038bcff5c5134d7e0
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(subst .,-,$(PKG_VERSION))
```
Comment (pprindeville) [inline utils/lm-sensors/Makefile]: If you're going to make this substitution twice, then either use a macro or a temporary variable.

Comment (dengqf6) [reply utils/lm-sensors/Makefile]: Done

**Comment by dengqf6** [comment]
@hnyman I think collectd's PKG_RELEASE should be bumped again because of ABI change

**Comment by hnyman** [comment]
I think that we should first set ABI_VERSION for lm-sensors, so that collectd would know exactly on which version it depends.

@jow-  has today implemented a better ABI dependency usage for packages. (work triggered by openssl version bump plus similar trouble e.g. with curl & ustreamssl earlier). See four recent  commits inhttps://github.com/openwrt/openwrt/commits/e6bcf1e4acc4901c1803cdccc4b9fabfd6d84c90

**Comment by dengqf6** [comment]
@hnyman 
I see, so in this case the value should be 5

**Pull request merged**